tasks/quality_gates.py

```diff
@@ -141,3 +144,77 @@ def parse_and_trigger_gates(ctx, config_path="test/static/static_quality_gates.y
     branch = os.environ["CI_COMMIT_BRANCH"]
     if github.get_pr_for_branch(branch).totalCount > 0:
         display_pr_comment(ctx, final_state == "success", gate_states, metric_handler)
+    # Generate PR to update static quality gates threshold once per day (scheduled main pipeline by conductor)
+    DDR_WORKFLOW_ID = os.environ.get("DDR_WORKFLOW_ID")
+    if DDR_WORKFLOW_ID and branch == "main" and is_conductor_scheduled_pipeline():
+        pr_url = update_quality_gates_threshold(ctx, metric_handler, github)
+        notify_threshold_update(pr_url)
+
+
+def get_gate_new_limit_threshold(current_gate, current_key, max_key, metric_handler):
+    # The new limit is decreased when the difference between current and max value is greater than the `BUFFER_SIZE`
+    curr_size = metric_handler.metrics[current_gate][current_key]
+    max_curr_size = metric_handler.metrics[current_gate][max_key]
+    remaining_allowed_size = max_curr_size - curr_size
+    gate_limit = max_curr_size
+    if remaining_allowed_size > BUFFER_SIZE:
+        saved_amount = remaining_allowed_size - BUFFER_SIZE
+        gate_limit -= saved_amount
+        metric_handler.total_size_saved += saved_amount
```

Comment:
This is incrementing a field that doesn't appear to exist, I suppose it's default initialized to 0 and behaves fine, but it feels like we're piggy backing a local state on an existing object that is not meant for that
I don't think it's blocking but it could be refactored to return a dict or a tuple with the calculation results